common/src/main/java/org/apache/uniffle/common/util/ChecksumUtils.java

```diff
@@ -40,14 +40,20 @@ public static long getCrc32(byte[] buf, int offset, int length) {
     return crc32.getValue();
   }
 
-  // you may need to flip at first
   public static long getCrc32(ByteBuffer byteBuffer) {
-    if (byteBuffer.hasArray()) {
-      return getCrc32(byteBuffer.array(), byteBuffer.arrayOffset() + byteBuffer.position(), byteBuffer.remaining());
-    } else {
-      byte[] byteArray = new byte[byteBuffer.remaining()];
-      byteBuffer.get(byteArray);
-      return getCrc32(byteArray);
+    return getCrc32(byteBuffer, byteBuffer.position(), byteBuffer.limit() - byteBuffer.position());
+  }
+
+  public static long getCrc32(ByteBuffer byteBuffer, int offset, int length) {
+    CRC32 crc32 = new CRC32();
+    ByteBuffer crcBuffer = byteBuffer.duplicate();
+    crcBuffer.position(offset);
+    for (int i = 0; i < length; ) {
+      int len = Math.min(LENGTH_PER_CRC, length - i);
+      crcBuffer.limit(crcBuffer.position() + len);
+      crc32.update(crcBuffer);
+      i += len;
```

Comment:
Spark has similar implement https://github.com/apache/spark/pull/33451/files#diff-8ada3fbd5ebaae4bc80daf73fab086971d1e8496f7e882e7510b1b15b389ff50
If we calculate all the data, I guess it may cost more memory.